analyzers/src/SonarAnalyzer.Common/Helpers/KnownAssembly.cs

```diff
@@ -39,16 +39,26 @@ public sealed partial class KnownAssembly
         new(And(NameIs("Microsoft.VisualStudio.QualityTools.UnitTestFramework").Or(NameIs("Microsoft.VisualStudio.TestPlatform.TestFramework")),
                 PublicKeyTokenIs("b03f5f7f11d50a3a")));
 
-    public static KnownAssembly FluentAssertions { get; } = new(NameIs("FluentAssertions").And(PublicKeyTokenIs("33f2691a05b67b6a")));
     public static KnownAssembly NFluent { get; } = new(NameIs("NFluent").And(OptionalPublicKeyTokenIs("18828b37b84b1437")));
-    public static KnownAssembly NSubstitute { get; } = new(NameIs("NSubstitute").And(PublicKeyTokenIs("92dd2e9066daa5ca")));
+    public static KnownAssembly FluentAssertions { get; } = new("FluentAssertions", "33f2691a05b67b6a");
+    public static KnownAssembly NSubstitute { get; } = new("NSubstitute", "92dd2e9066daa5ca");
+    // Logging assemblies
+    public static KnownAssembly MicrosoftExtensionsLoggingAbstractions { get; } = new("Microsoft.Extensions.Logging.Abstractions", "adb9793829ddae60");
+    public static KnownAssembly Serilog { get; } = new("Serilog", "24c2f752a8e58a10");
+    public static KnownAssembly NLog { get; } = new("NLog", "5120e14c03d0593c");
+    public static KnownAssembly Log4Net { get; } = new("log4net", "669e0ddf0bb1aa2a");
+    public static KnownAssembly CommonLoggingCore { get; } = new("Common.Logging.Core", "af08829b84f0328e");
+    public static KnownAssembly CastleCore { get; } = new("Castle.Core", "407dd0808d44fbdc");
+
+    internal KnownAssembly(string name, string publicKeyToken)
+        : this(NameIs(name).And(PublicKeyTokenIs(publicKeyToken)))
+    { }
```

Comment:
Please remove. This is a fluent API and should be used like that.

Reply:
Because I do not see any repetition:
https://github.com/SonarSource/sonar-dotnet/blob/f05a95b37b13ffc9a4af167b05590b52bd3bf042/analyzers/src/SonarAnalyzer.Common/Helpers/KnownAssembly.cs#L29-L44

and you already need multiple public keys for e.g. log4net:
https://nuget.info/packages/log4net/1.2.10  -> 669e0ddf0bb1aa2a
https://nuget.info/packages/log4net/2.0.15 -> 1b44e1d426115821

Reply:
What you can do is add a predicate "NameAndPublicKeyIs".